Code/Framework/AzCore/AzCore/Debug/PerformanceCollector.h

```diff
@@ -0,0 +1,178 @@
+        //! is already in effect.
+        void UpdateWaitTimeBeforeEachBatch(AZStd::chrono::seconds seconds);
+
+        //! Calling this one with newValue > 0 will trigger json file creation
+        //! and performance capture for as many batches.
+        void UpdateNumberOfCaptureBatches(AZ::u32 newValue);
+
+        const AZ::IO::Path& GetOutputFilePath() { return m_outputFilePath;  }
+
+        const AZStd::string& GetOutputDataBuffer() { return m_outputDataBuffer; }
+    
+    private:
+        //! A helper function that loops across all statistics in @m_statisticsManager
+        //! and reports each result into @m_eventLogger.
+        void RecordStatistics();
+
+        void RestartPeriodicEventStamps();
+
+        void CreateOutputJsonFile();
+
+        //! Main control parameters. Usually mirrored by CVARs. They can change at runtime.
+        AZ::u32 m_numberOfCaptureBatches = 0; // A number greater than 0 starts performance collection.
+        DataLogType m_dataLogType = DataLogType::LogStatistics; // Defines the data collection and report mode.
+        AZ::u32 m_frameCountPerCaptureBatch = 50; // How many frames of data will be capture per batch.
+        AZStd::chrono::seconds m_waitTimeBeforeEachBatch = AZStd::chrono::seconds(3); // How many seconds to wait before starting the next batch of data capture.
+
+        AZStd::string m_logCategory;
+        AZ::u32 m_frameCountInCurrentBatch = 0; // How many frames have been captured so far within the current batch.
+        AZStd::chrono::steady_clock::time_point m_startWaitTime;
+        bool m_isWaitingBeforeNextBatch = true;
+        OnBatchCompleteCallback m_onBatchCompleteCallback; // A notification will be sent each time a batch of frames is performance collected.
+   
+        //! Only used when @m_captureType == CaptureType::LogStatistics.
+        AZ::Statistics::StatisticsManager<AZStd::string> m_statisticsManager;
+    
+        //! Only used to store the previous value when RecordPeriodicEvent() is called
+        //! for any given metrics.
+        AZStd::unordered_map<AZStd::string, AZStd::chrono::steady_clock::time_point> m_periodicEventStamps;
+
+        //! In some circumstances, like running under UnitTests, an output file won't be created.
+        //! Instead the output data will be streamed into this buffer.
+        AZStd::string m_outputDataBuffer;
+
+        AZ::IO::Path m_outputFilePath; // We store here the file path of the most recently created output file.
+        Metrics::JsonTraceEventLogger m_eventLogger;
+    }; // class PerformanceCollector
+    
+    //! A Convenience class used to measure time performance of scopes of code
+    //! with constructor/destructor.
+    class ScopeDuration
+    {
+    public:
+        ScopeDuration() = delete;
+    
+        explicit ScopeDuration(PerformanceCollector* performanceCollector, const AZStd::string_view metricName)
```

Comment:
`explicit` is not needed for a function with 2 required parameters.
```suggestion
        ScopeDuration(PerformanceCollector* performanceCollector, const AZStd::string_view metricName)
```

Reply:
Fixed.